HabitTracker.GreyKlouds/HabitTracker/Program.cs

```diff
@@ -0,0 +1,248 @@
+                    tableData.Add
+                    (
+                        new DrinkingWater
+                        {
+                            Id = reader.GetInt32(0),
+                            Date = DateTime.ParseExact(reader.GetString(1), "mm-dd-yy", new CultureInfo("en-US")),
+                            Quantity = reader.GetInt32(2)
+                        }
+                    );
+                }
+            }
+            else
+            {
+                Console.WriteLine("No rows found");
+            }
+
+            connection.Close();
+
+            Console.WriteLine("-----------------------------------------");
+            foreach (var dw in tableData)
+            {
+                Console.WriteLine($"{dw.Id} - {dw.Date.ToString("mm-dd-yyyy")} - Quantity: {dw.Quantity}");
+            }
+            Console.WriteLine("-----------------------------------------");
+        }
+
+    }
+    private static void Delete()
+    {
+        Console.Clear();
+        GetAllRecords();
+
+        var recordId = GetNumberInput("\n\nPlease select a record you wish to delete. Press 'n' to return to the Main Menu.");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText = $"Delete from DrinkingWater Where Id = '{recordId}'";
+
+            int rowCount = tableCmd.ExecuteNonQuery();
+            if (rowCount == 0)
+            {
+                Console.WriteLine($"\n\nRecord with Id {recordId} does not exist.");
+                Delete();
+            }
+        }
+        Console.WriteLine($"\n\nRecord withd Id {recordId} has been deleted.\n\n");
+
+        GetUserinput();
+    }
+    internal static int GetNumberInput(string message)
+    {
+        Console.WriteLine(message);
+
+        string numberInput = Console.ReadLine();
+
+        if (numberInput == "0") GetUserinput();
+
+        while (!Int32.TryParse(numberInput, out _) || Convert.ToInt32(numberInput) < 0)
+        {
+            Console.WriteLine("\n\nInvalid number.Please try again:");
+            numberInput = Console.ReadLine();
+        }
+
+        int finalInput = Convert.ToInt32(numberInput);
+
+        return finalInput;
+    }
+    internal static string GetDateInput()
+    {
+        Console.WriteLine("\n\nPlease insert the date: (Format: mm-dd-yy). Type 0 to return to main menu.");
+
+        string dateInput = Console.ReadLine();
+
+        if (dateInput == "0") GetUserinput();
+
+        while (!DateTime.TryParseExact(dateInput, "mm-dd-yy", new CultureInfo("en-US"), DateTimeStyles.None, out _))
+        {
+            Console.WriteLine("\n\nInvalid date. (Format: mm-dd-yy). Type 0 to return to main manu or try again:\n\n");
+            dateInput = Console.ReadLine();
+        }
+
+        return dateInput;
+    }
+}
+public class DrinkingWater

```

Comment:
🟠 **Object Orientated Programming**

💡 Again, I know the project scope allows for this. But I would really of liked to see this in it's own file `HabitRecord.cs`

🔧 Refactor into it's own file.